MetadataExtractor/IO/RandomAccessStream.cs

```diff
@@ -141,7 +152,20 @@ public int Read(long index, byte[] buffer, int offset, int count, bool isSequent
         /// <exception cref="IOException"/>
         public byte GetByte(long index, bool isSequential)
         {
-            ValidateIndex(index, 1, isSequential);
+            return GetByte(index, isSequential, true);
+        }
+
+        /// <summary>Returns an unsigned byte at an index in the sequence.</summary>
+        /// <returns>the 8 bit int value, between 0 and 255</returns>
+        /// <param name="index">position within the data buffer to read byte</param>
+        /// <param name="isSequential">flag indicating if caller is using sequential access</param>
+        /// <param name="validateIndex">allows for skipping validation if already done by the caller</param>
+        /// <exception cref="BufferBoundsException"/>
+        /// <exception cref="IOException"/>
+        private byte GetByte(long index, bool isSequential, bool validateIndex)
```

Comment:
Given how often we will call this method I think you could make this method `GetByteNoValidation` and move the validation back to `GetByte`, then remove the `validateIndex` parameter.

Reply:
Implemented, with some other changes